src/factories/guide.ts

```diff
@@ -0,0 +1,73 @@
+import { Container } from 'pixi.js'
+import { rectangleFactory } from '@/factories/rectangle'
+import { FormatDate } from '@/models/guides'
+import { waitForViewport } from '@/objects'
+import { waitForApplication } from '@/objects/application'
+import { waitForConfig } from '@/objects/config'
+import { waitForCull } from '@/objects/culling'
+import { waitForFonts } from '@/objects/fonts'
+import { waitForScale } from '@/objects/scale'
+
+export type GuideFactory = Awaited<ReturnType<typeof guideFactory>>
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export async function guideFactory() {
+  const application = await waitForApplication()
+  const viewport = await waitForViewport()
+  const cull = await waitForCull()
+  const config = await waitForConfig()
```

Comment:
@pleek91 in this scenario, can I expect config to always provide the latest values? Or should this be moved into the render functions?

Reply:
since config is an object every time you access it like config.xyz it will always be the correct value. You just cannot destructure it and use the value later or you'll likely end up with the same value each time. Similar to breaking reactivity in vue

Reply:
Gotchya, appreciate it.